pylint/checkers/typecheck.py

```diff
@@ -1145,8 +1155,10 @@ def visit_with(self, node):
                 # manager and give up, otherwise emit not-context-manager.
                 # See the test file for not_context_manager for a couple
                 # of self explaining tests.
-                for path in filter(None, _unflatten(context.path)):
-                    scope = path.scope()
+                context_path_names = (filter(None, _unflatten(context.path)))
+                inferred_paths = _flatten_container(path.infer() for path in context_path_names)
```

Comment:
Might be worth adding a comment on what we are inferring inside the `context_path_names`